environment_gpu.yaml

```
@@ -1,23 +1,20 @@
 name: sam
 channels:
-    - pytorch
-    - nvidia
     - conda-forge
 dependencies:
     - imagecodecs
-    # This pin is necessary because later nifty versions have import errors on windows.
-    - nifty =1.2.1=*_4
+    - nifty >=1.2.1
     - magicgui
     - napari >=0.5.0
     - natsort
     - pip
     - pooch
-    - protobuf <5
     - pyqt
     - python-xxhash
     - python-elf >=0.4.8
-    - pytorch >=2.4
-    - pytorch-cuda>=11.7  # you may need to update the cuda version to match your system
+    # This does not work for me.
+    # - pytorch >=2.4 "libtorch=*=cuda*"
+    - pytorch-gpu
```

```suggestion
    - pytorch >=2.4 
    - libtorch=*=cuda*
    # - pytorch-gpu
```

libtorch is a dependent package of pytorch. They are designed to match cpu/cpu cuda/cuda.

it is difficult (impossible maybe????) to write a constraint that says "greater than this version, with this build number string". 

So I use a trick, where one package specifies the build version, the other specifies the build string I want.